File: BWHJsInterop.cs
using Microsoft.AspNetCore.Components;
using Microsoft.JSInterop;
using System;
using System.Threading.Tasks;

namespace BlazorWindowHelper
{
    public class BWHJsInterop
    {


        public static ValueTask<string> Prompt(string message)
        {

            BWHelperFunctions.CheckHasJsRuntime();

            return BWHWindowHelper.jsRuntime.InvokeAsync<string>(
                "BWHJsFunctions.showPrompt",
                message);
        }

        public static ValueTask<bool> Alert(string message)
        {

            BWHelperFunctions.CheckHasJsRuntime();

            return BWHWindowHelper.jsRuntime.InvokeAsync<bool>(
                "BWHJsFunctions.alert",message);
            
        }


        public static ValueTask<bool> Print()
        {

            BWHelperFunctions.CheckHasJsRuntime();

            return BWHWindowHelper.jsRuntime.InvokeAsync<bool>(
                "BWHJsFunctions.Print");

        }

        public static ValueTask SetFocus(string id)
        {
            BWHelperFunctions.CheckHasJsRuntime();
            return BWHWindowHelper.jsRuntime.InvokeVoidAsync(
                "BWHJsFunctions.SetFocus", id);
        }



        public static ValueTask<bool> JsLog(string message)
        {
            BWHelperFunctions.CheckHasJsRuntime();

            return BWHWindowHelper.jsRuntime.InvokeAsync<bool>(
                "BWHJsFunctions.mylog", message);

        }

        public static ValueTask<bool> LogWithTime(string message)
        {

            BWHelperFunctions.CheckHasJsRuntime();

            return BWHWindowHelper.jsRuntime.InvokeAsync<bool>(
                "BWHJsFunctions.logWithTime", message);

        }

        public static ValueTask<bool> SetOnOrOff(bool OnOrOff)
        {
            BWHelperFunctions.CheckHasJsRuntime();

            return BWHWindowHelper.jsRuntime.InvokeAsync<bool>(
                "BWHJsFunctions.setOnOrOff", OnOrOff);

        }

        public static ValueTask<int> GetTimezoneOffset()
        {
            BWHelperFunctions.CheckHasJsRuntime();

            return BWHWindowHelper.jsRuntime.InvokeAsync<int>(
                "BWHJsFunctions.GetTimezoneOffset");
        }

        public static ValueTask<long> GetDateMilliseconds()
        {
            BWHelperFunctions.CheckHasJsRuntime();

            return BWHWindowHelper.jsRuntime.InvokeAsync<long>(
                "BWHJsFunctions.GetDateMilliseconds");
        }

        public static ValueTask<double> GetElementActualWidth(string elementID)
        {
            BWHelperFunctions.CheckHasJsRuntime();
            return BWHWindowHelper.jsRuntime.InvokeAsync<double>(
                "BWHJsFunctions.GetElementActualWidth", elementID);
        }

        public static ValueTask<double> GetElementActualHeight(string elementID)
        {
            BWHelperFunctions.CheckHasJsRuntime();
            return BWHWindowHelper.jsRuntime.InvokeAsync<double>(
                "BWHJsFunctions.GetElementActualHeight", elementID);
        }

        public static ValueTask<double> GetElementActualTop(string elementID)
        {
            BWHelperFunctions.CheckHasJsRuntime();
            return BWHWindowHelper.jsRuntime.InvokeAsync<double>(
                "BWHJsFunctions.GetElementActualTop", elementID);
        }

        public static ValueTask<double> GetElementActualLeft(string elementID)
        {
            BWHelperFunctions.CheckHasJsRuntime();
            return BWHWindowHelper.jsRuntime.InvokeAsync<double>(
                "BWHJsFunctions.GetElementActualLeft", elementID);
        }
   
        public static void ClearConsole()
        {
            BWHelperFunctions.CheckHasJsRuntime();
            BWHWindowHelper.jsRuntime.InvokeVoidAsync(
                "BWHJsFunctions.ClearConsole");
        }


        public static void SetElementTop(string elementID, int t)
        {
            BWHelperFunctions.CheckHasJsRuntime();
            BWHWindowHelper.jsRuntime.InvokeVoidAsync(
                "BWHJsFunctions.SetElementTop", elementID, t);
        }

        public static void SetElementLeft(string elementID, int l)
        {
            BWHelperFunctions.CheckHasJsRuntime();
            BWHWindowHelper.jsRuntime.InvokeVoidAsync(
                "BWHJsFunctions.SetElementLeft", elementID, l);
        }

        public static void SetElementWidth(string elementID, int w)
        {
            BWHelperFunctions.CheckHasJsRuntime();
            BWHWindowHelper.jsRuntime.InvokeVoidAsync(
                "BWHJsFunctions.SetElementWidth", elementID, w);
        }

        public static void SetElementHeight(string elementID, int h)
        {
            BWHelperFunctions.CheckHasJsRuntime();
            BWHWindowHelper.jsRuntime.InvokeVoidAsync(
                "BWHJsFunctions.SetElementHeight", elementID, h);
        }

        public static ValueTask<double> GetWindowWidth()
        {
            BWHelperFunctions.CheckHasJsRuntime();
            return BWHWindowHelper.jsRuntime.InvokeAsync<double>(
                "BWHJsFunctions.GetWindowWidth");
        }

        public static ValueTask<double> GetWindowHeight()
        {
            BWHelperFunctions.CheckHasJsRuntime();
            return BWHWindowHelper.jsRuntime.InvokeAsync<double>(
                "BWHJsFunctions.GetWindowHeight");
        }


       
    }
}

File: wwwroot/BWHJsInterop.js
var OnOrOff = false;

document.onkeyup = function (evt) {

    if (OnOrOff === true) {
        evt = evt || window.event;
        DotNet.invokeMethodAsync('BlazorWindowHelper', 'InvokeKeyUp', [evt.keyCode, evt.ctrlKey, evt.shiftKey, evt.altKey]);
    }
};


document.onkeydown = function (evt) {

    if (OnOrOff === true) {
        evt = evt || window.event;
        DotNet.invokeMethodAsync('BlazorWindowHelper', 'InvokeKeyDown', [evt.keyCode, evt.ctrlKey, evt.shiftKey, evt.altKey]);
    }
};


document.addEventListener('visibilitychange', onVisibilityChange, false);


window.addEventListener("scroll", onScroll, false);
window.addEventListener("resize", onResize, false);
window.addEventListener("unload", onUnload, false);


function onScroll() {
    if (OnOrOff === true) {
        DotNet.invokeMethodAsync('BlazorWindowHelper', 'InvokeOnScroll');
    }
}

function onResize() {
    if (OnOrOff === true) {
        DotNet.invokeMethodAsync('BlazorWindowHelper', 'InvokeOnResize');
    }
}

function onUnload() {
    if (OnOrOff === true) {
        DotNet.invokeMethodAsync('BlazorWindowHelper', 'InvokeOnUnload');
    }
}


function onVisibilityChange() {
    if (OnOrOff === true) {
            DotNet.invokeMethodAsync('BlazorWindowHelper', 'InvokeOnVisibilityChange', !document.hidden);
    }
}

window.BWHJsFunctions = {
    showPrompt: function (message) {
    return prompt(message, 'Type anything here');
    },
    ClearConsole: function (message) {
        console.clear();
        return true;
    },
    alert: function (message) {
        alert(message);
        return true;
    },
    Print: function () {
        window.print();
        return true;
    },
    mylog: function (message) {
        console.log(message);
        return true;
    },
    logWithTime: function (message) {
        console.log(getTime() +" " + message);
        return true;
    },
    SetFocus: function (id) {
        if (document.getElementById(id) !== null) {
            document.getElementById(id).focus();
        }
    },
    setOnOrOff: function (b) {
        OnOrOff = b;
        return true;
    },
    GetTimezoneOffset: function () {
        return new Date().getTimezoneOffset() / 60;
    },
    GetDateMilliseconds: function () {
        return new Date().getTime();
    },
    GetElementActualWidth: function (el) {
        if (document.getElementById(el) !== null) {
            let rect = document.getElementById(el).getBoundingClientRect();
            return rect.width;
        }
        else {
            return 0;
        }
    },
    GetElementActualHeight: function (el) {
        if (document.getElementById(el) !== null) {
            let rect = document.getElementById(el).getBoundingClientRect();
            return rect.height;
        }
        else {
            return 0;
        }
    },
    GetElementActualTop: function (el) {
        if (document.getElementById(el) !== null) {
            let rect = document.getElementById(el).getBoundingClientRect();
            return rect.top;
        }
        else {
            return 0;
        }
    },
    GetElementActualLeft: function (el) {
        if (document.getElementById(el) !== null) {
            let rect = document.getElementById(el).getBoundingClientRect();
            return rect.left;
        }
        else {
            return 0;
        }
    },
    SetElementTop: function (el, t) {
        if (document.getElementById(el) !== null) {
            document.getElementById(el).style.top = t + "px";
        }
    },
    SetElementLeft: function (el, l) {
        if (document.getElementById(el) !== null) {
            document.getElementById(el).style.left =l+"px";
        }
    },
    SetElementWidth: function (el, w) {
        if (document.getElementById(el) !== null) {
            document.getElementById(el).style.width = w + "px";
        }
    },
    SetElementHeight: function (el, h) {
        if (document.getElementById(el) !== null) {
            document.getElementById(el).style.height = h + "px";
        }
    },
    GetWindowWidth: function () {
        return window.innerWidth
            || document.documentElement.clientWidth
            || document.body.clientWidth;

    },
    GetWindowHeight: function () {
        return window.innerHeight
            || document.documentElement.clientHeight
            || document.body.clientHeight;

    },
    SetItemLocalStorage: function (k, v) {
        localStorage.setItem(k, v);
        return true;
    },
    GetItemLocalStorage: function (k) {
        return localStorage.getItem(k);
    },
    RemoveItemLocalStorage: function (k) {
        var v = localStorage.getItem(k);
        if (v === null) {
            return false;
        }
        else {
            localStorage.removeItem(k);
            return true;
        }
    },
    CheckIfKeyExistsLocalStorage: function (k) {
        var v = localStorage.getItem(k);
        if (v === null) {
            return false;
        }
        else {
            return true;
        }
    },
    ClearLocalStorage: function () {
        localStorage.clear();
        return true;
    },
};
